File: Overoom.React/src/services/room-hub/events/notifications/targeted-notification.event.ts
import { NotificationEvent } from './notification.event.ts';

/** Базовое событие уведомления, направленное на конкретного пользователя. */
export interface TargetedNotificationEvent extends NotificationEvent {
  /** Идентификатор пользователя, на которого направлено событие. */
  target: string;
}

File: Rooms.Application.Abstractions/Events/Notifications/TargetedNotificationEvent.cs
namespace Rooms.Application.Abstractions.Events.Notifications;

/// <summary>
/// Базовое событие уведомления, направленное на конкретного пользователя.
/// </summary>
public abstract class TargetedNotificationEvent : NotificationEvent
{
    /// <summary>
    /// Идентификатор пользователя, на которого направлено событие.
    /// </summary>
    public required Guid Target { get; init; }
}